openest/generate/functions.py

```diff
@@ -443,6 +443,148 @@ def describe():
+        for ii in range(1, len(subcalcs)):
+            assert subcalcs[0].unitses[0] == subcalcs[ii].unitses[0], "%s <> %s" % (subcalcs[0].unitses[0], subcalcs[ii].unitses[0])
+            fullunitses.extend(subcalcs[ii].unitses)
+        if unshift:
+            super().__init__([subcalcs[0].unitses[0]] + fullunitses)
+        else:
+            super().__init__([subcalcs[0].unitses[0]])
+
+        self.unshift = unshift
+        self.subcalcs = subcalcs
+        self.frac = frac
+
+
+    def format(self, lang, *args, **kwargs):
+        raise NotImplementedError
+        mains = []
+        elements = {}
+        alldeps = set()
+        for subcalc in self.subcalcs:
+            elements.update(subcalc.format(lang, *args, **kwargs))
+            mains.append(elements['main'])
+            alldeps.update(elements['main'].dependencies)
+
+        if lang in ['latex', 'julia']:
+            elements['main'] = FormatElement(' * '.join([main.repstr for main in mains]), list(alldeps))
+
+        formatting.add_label('product', elements)
+        return elements
+
+    def apply(self, region, *args, **kwargs):
+        """Apply calculation to all subcalculations
+
+        All parameters are passed to ``self.subcalc.apply()``.
+
+        Parameters
+        ----------
+        region : str
+        args
+        kwargs
+
+        Returns
+        -------
+        openest.generate.Calculation.ApplicationPassCall
+        """
+
+        def generate(year, results):
+            frac, x1, x2 = (r[1] for r in results)
+
+            if frac > 1 or frac < 0:
+                raise ValueError("`frac` results must be within [0, 1]")
+
+            return frac * x1 + (1 - frac) * x2
+
+        calcs = [self.frac, *self.subcalcs]
+        # Prepare the generator from our encapsulated operations
+        subapps = [c.apply(region, *args, **kwargs) for c in calcs]
+        return calculation.ApplicationPassCall(region, subapps, generate, unshift=self.unshift)
+
+    def column_info(self):
+        """Get column information of values output from this calculation.
+
+        Returns
+        -------
+        Sequence of dicts
+            Each dict contains:
+
+                ``"name"``
+                    Short-length title of this calculation
+
+                ``"title"``
+                    Long-length title of this calculation
+
+                ``"description"``
+                    Long-form description of this calculation
+        """
+        infoses = [subcalc.column_info() for subcalc in self.subcalcs]
+        title = 'FractionSum of previous results'
+        description = 'FractionSum of ' + ', '.join([infos[0]['title'] for infos in infoses])
+        fullinfos = [info for infos in infoses for info in infos]
+        return [dict(name='fractionsum', title=title, description=description)] + fullinfos
+
+    def enable_deltamethod(self):
+        raise NotImplementedError
+
+    def partial_derivative(self, covariate, covarunit):
```

Comment:
Any reason you're not implementing this?

Reply:
`Product` isn't merged into `master` yet. I'll get on it as soon as it's in.

Reply:
Resolved with 20ecad1, should be checked.